src/run.rs

```diff
@@ -0,0 +1,25 @@
+use actix_web::{dev::Server, web::Data, App, HttpServer};
+use anyhow::{Context, Result};
+
+use log::info;
+
+use crate::router::router;
+use crate::state::State;
+
+pub fn run(address: &str) -> Result<Server> {
```

Comment:
Please make use of `SocketAddr` for this, lets avoind using primitive data types when we have dedicated types available.

This way we would have parameter validation by design.

Refer to: https://doc.rust-lang.org/std/net/enum.SocketAddr.html

Reply:
I changed to TcpListener, that allows us to use random ports to spawn the application in the integration tests